content/doc/developer/publishing/documentation.adoc

```diff
@@ -0,0 +1,169 @@
+maintainers to provide the same documentation from GitHub pages and the
+Jenkins plugin site, and at the same time it allows using the
+Documentation-as-Code techniques when the documentation is a part of the
+repository and hence all common practices can be applied (versioning,
+pull requests and reviews, creating documentation in parallel with
+features, editing docs from GitHub Web UI).
+
+The plugin site can pull documentation from the root README pages and
+from other locations. For new plugins it is recommended to use README
+pages as a source of documentation, preferably in Markdown or Asciidoc
+format (TXT is also supported in such case).
+
+How to publish the plugin documentation from GitHub?
+
+* Create a README page and put the plugin documentation there
+** This page will become a landing page for https://plugins.jenkins.io/
+** More documentation pages can be introduced inside the repository and
+linked from the README, the plugin site will display both absolute and
+relative links
+** Images from pages will be displayed by the plugin site as well
+* Modify your project URL to point to the GitHub repository,
+e.g. http://github.com/jenkinsci/your-plugin
+** See the guidelines for Maven and Gradle below
+* Once the new plugin version is released, link:https://plugins.jenkins.io/[Plugin site] with start using it as a documentation source
+
+Documentation examples:
+
+* https://plugins.jenkins.io/configuration-as-code
+* https://plugins.jenkins.io/gradle
+* https://plugins.jenkins.io/mailer
+
+=== Using Jenkins Wiki as source of documentation
+
+Even though for new plugins it is recommended to just use GitHub README files, 
+link:https://wiki.jenkins.io[Jenkins Wiki] can still be used if there are strong reasons to do so.
+See link:../wiki-page[this page] for more information.
+
+=== Referencing the documentation page from the project file
+
+You should link to your plugin's documentation, whether on the wiki or
+elsewhere, in your plugin's pom.xml, like this:
+
+```xml
+<project>
+  ...
+  <url>http://github.com/jenkinsci/your-plugin</url>
+  ...
+</project>
+```
+
+If you're building your plugin with https://github.com/jenkinsci/gradle-jpi-plugin[Gradle], 
+you can set the URL in your `+build.gradle+` like so:
+
+```groovy
+jenkinsPlugin {
+  // ...
+  url = 'http://github.com/jenkinsci/your-plugin'
+  // ...
+}
+```
+
+Or if you have a plugin written in
+https://wiki.jenkins.io/display/JENKINS/Jenkins+plugin+development+in+Ruby[Ruby],
+you must edit your `+.pluginspec+` file like so:
+
+```ruby
+Jenkins::Plugin::Specification.new do |plugin|
+  # ...
+  plugin.url = 'http://github.com/jenkinsci/your-plugin'
+  # ...
+end
+```
+
+This ensures that the update center will list your plugin correctly once
+the new plugin version is released. If this is missing, or does not
+point to your Jenkins wiki page, your plugin will not be included in the
+update center.
+
+== Adding Maintainer Information
```

Comment:
Pretty sure this will be obsolete soon, so I don't think it's helpful to add this.